quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/DebugHostnameSettingsResource.java

```diff
@@ -0,0 +1,144 @@
+import org.keycloak.theme.freemarker.FreeMarkerProvider;
+import org.keycloak.urls.UrlType;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+@Path("/realms")
+public class DebugHostnameSettingsResource {
+    public static final String DEFAULT_PATH_SUFFIX = "hostname-debug";
+
+    @Context
+    private KeycloakSession keycloakSession;
+
+    private final Map<String, String> allConfigPropertiesMap;
+
+    public DebugHostnameSettingsResource() {
+
+        this.allConfigPropertiesMap = new LinkedHashMap<>();
+        for (String key : ConstantsDebugHostname.RELEVANT_OPTIONS) {
+            addOption(key);
+        }
+
+    }
+
+    @GET
+    @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX)
+    @Produces(MediaType.TEXT_HTML)
+    public String debug(final @PathParam("realmName") String realmName) throws IOException, FreeMarkerException {
+        FreeMarkerProvider freeMarkerProvider = keycloakSession.getProvider(FreeMarkerProvider.class);
+        RealmModel realmModel = keycloakSession.realms().getRealmByName(realmName);
+
+        URI frontendUri = keycloakSession.getContext().getUri(UrlType.FRONTEND).getBaseUri();
+        URI backendUri = keycloakSession.getContext().getUri(UrlType.BACKEND).getBaseUri();
+        URI adminUri = keycloakSession.getContext().getUri(UrlType.ADMIN).getBaseUri();
+
+        String frontendTestUrl = getTest(realmModel, frontendUri);
+        String backendTestUrl = getTest(realmModel, backendUri);
+        String adminTestUrl = getTest(realmModel, adminUri);
+
+        Map<String, Object> attributes = new HashMap<>();
+        attributes.put("frontendUrl", frontendUri.toString());
+        attributes.put("backendUrl", backendUri.toString());
+        attributes.put("adminUrl", adminUri.toString());
+
+        attributes.put("realm", realmModel.getName());
+        attributes.put("realmUrl", realmModel.getAttribute("frontendUrl"));
+
+        attributes.put("frontendTestUrl", frontendTestUrl);
+        attributes.put("backendTestUrl", backendTestUrl);
+        attributes.put("adminTestUrl", adminTestUrl);
+
+        attributes.put("serverMode", Environment.isDevMode() ? "dev [start-dev]" : "production [start]");
+
+        attributes.put("config", this.allConfigPropertiesMap);
+        attributes.put("headers", getHeaders());
+
+        return freeMarkerProvider.processTemplate(
+                attributes,
+                "debug-hostname-settings.ftl",
+                keycloakSession.theme().getTheme("base", Theme.Type.LOGIN)
+        );
+    }
+
+    @GET
+    @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX + "/test-cors-in-hearders")
```

Comment:
There's a typo in `hearders`. I think for simplicity, we could keep it as
```suggestion
    @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX + "/test")
```